src/main/java/Hachi.java

```diff
@@ -0,0 +1,266 @@
+            String fromDate = line.substring(indexOfStart, indexOfEnd - 3).trim();
+            String toDate = line.substring(indexOfEnd).trim();
+            toAdd = new Event(name, fromDate, toDate);
+        } else {
+            toAdd = new Task(line);
+        }
+        int numTasks = Task.getTotalNumTasks();
+        listOfTasks[numTasks - 1] = toAdd;
+        System.out.println("    Added to list: " + toAdd.getName());
+    }
+
+    /**
+     * Given a task's index and the list of tasks,
+     * mark that task as complete.
+     * @param index Index of the task to be marked.
+     * @param listOfTasks The Task[] array that contains the list of tasks.
+     */
+
+    public static void markTask(int index, Task[] listOfTasks) {
+        listOfTasks[index].setCompleteness(true);
+        System.out.println("    Sure, I've marked this task as done:");
+        String statusIcon = listOfTasks[index].getStatusIcon();
+        System.out.print("    " + (index + 1) + ": ");
+        System.out.print("[" + statusIcon + "] ");
+        System.out.println(listOfTasks[index].getName());
+    }
+
+    /**
+     * Given a task's index and the list of tasks,
+     * mark that task as incomplete.
+     *
+     * @param index Index of the task to be marked.
+     * @param listOfTasks The Task[] array that contains the list of tasks.
+     */
+
+    public static void unmarkTask(int index, Task[] listOfTasks) {
+        listOfTasks[index].setCompleteness(false);
+        System.out.println("    Okay, marking this task as incomplete now:");
+        String statusIcon = listOfTasks[index].getStatusIcon();
+        System.out.print("    " + (index + 1) + ": ");
+        System.out.print("[" + statusIcon + "] ");
+        System.out.println(listOfTasks[index].getName());
+    }
+
+    /**
+     * Prints to the console a goodbye message for the user.
+     */
+
+    public static void printGoodbyeMessage() {
+        System.out.println("    Goodbye! Hope you have a marvelous day.");
+        spacerInsert("medium", true);
+    }
+
+    /**
+     * The main program that starts the chatbot.
+     * Prints to the console for the user to read its messages.
+     * Greets the user and awaits user input.
+     * <p>
+     * Chatbot can:
+     * <p>1. retrieve list of tasks with user input "list"
+     * <p>2. mark or unmark tasks complete with user input "mark #tasknumber"
+     * <p>3. say goodbye to the user with user input "bye" or "goodbye"
+     * <p>4. add a to-do to the list of task with "todo <event name>"
+     * <p>5. add a deadline to the list of task with "deadline <event name> /by <by date>"
+     * <p>6. add an event to the list of task with "event <event name> /from <start date> /to <end date>"
+     * <p>7. retrieve a list of chatbot commands with "help"
+     *
+     * @param args Command line arguments - not used.
+     */
+
+    public static void main(String[] args) {
+        spacerInsert("medium", false);
```

Comment:
I would say for this, make sure you separate your declarations from the rest of the code (keep them on the top and add a one line space between running methods)